cli/src/main/java/org/jboss/as/cli/parsing/ParserUtil.java

```diff
@@ -238,10 +238,13 @@ public void leavingState(ParsingContext ctx) throws CommandFormatException {
                         }
                     } else {
                         final String value = buffer.toString().trim();
-                        if (ch == '=') {
+                        // If ctx.isEndOfContent() is true and the last character is '=' or ':', it must be escaped and
+                        // it won't be considered a type/name separator (if it wasn't escaped the NodeState would've
+                        // been left already).
+                        if (ch == '=' && !ctx.isEndOfContent()) {
                             handler.nodeType(bufferStartIndex, value);
                             handler.nodeTypeNameSeparator(ctx.getLocation());
-                        } else if (ch == ':') {
+                        } else if (ch == ':' && !ctx.isEndOfContent()) {
```

Comment:
@jfdenise thanks - glad I could help. I added the comment about ```ctx.isEndOfContent()``` and additionally handled also the ':' character in the same way. ':' was accidentally working, but basically the same case as '='.